core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/validators/StreamValidator.java

```diff
@@ -57,11 +54,15 @@ private void requireExistingDomain(Stream stream) {
   }
 
   private void requireExistingSchema(Stream stream) {
-    if (stream.getSchemaKey() == null) {
-      throw new ValidationException("Schema must be specified");
-    }
+    requireSchemaSpecified(stream);
     if (!schemaView.exists(stream.getSchemaKey())) {
       throw new ValidationException("Schema does not exist");
     }
   }
+
+  private void requireSchemaSpecified(Stream stream) {
+    if (stream.getSchemaKey() == null) {
+      throw new ValidationException("Schema must be specified");
+    }
+  }
```

Comment:
What is the goal of the changes in these validations? As far as I can tell all that has been changed is that previously, on update a schema key was required to be present and the schema exists. But now, it seems like on update the schema is required to be present, but is NOT required to exist? I don't think this is desirable?